### DISCUSSION TITLE
Profiling fiber-based job system using third-party instrumentation profiler

### DISCUSSION THREAD

**Discussion by Dragnalith** [nem0/LumixEngine#1426] [General]
Hello,

I wonder if anyone have experience about using thirdparty instrumentation profiler to profile Lumix Engine and its fiber-based job system?

I get used to fiber-based job system, but one drawback is all instrumentation profiler I know do not support fiber context switch (e.g Superluminal). They get confused because an EndEvent() closing a BeginEvent() on a thread is not guarantee to happen at the same stack level once thread jump from one fiber to another.

In that case, how do you do detailed profiling?

**Comment by nem0** [comment]
Hi, Tracy https://github.com/wolfpld/tracy supports fibers in the latest version. I've not tried though. I use only Lumix's profiler, which has proper support for its job system.

**Reply by Dragnalith** [reply]
thank you, I will investigate tracy!
